nav2_velocity_smoother/src/velocity_smoother.cpp

```diff
@@ -182,9 +182,21 @@ double VelocitySmoother::findEtaConstraint(
   const double v_curr, const double v_cmd, const double accel, const double decel)
 {
   // Exploiting vector scaling properties
-  const double v_component_max = accel / smoothing_frequency_;
-  const double v_component_min = decel / smoothing_frequency_;
-  const double dv = v_cmd - v_curr;
+  double dv = v_cmd - v_curr;
+
+  double v_component_max;
+  double v_component_min;
+
+  // Accelerating if magnitude of v_cmd is above magnitude of v_curr
+  // and if v_cmd and v_curr have the same sign (i.e. speed is NOT passing through 0.0)
+  // Deccelerating otherwise
+  if ((abs(v_cmd) >= abs(v_curr)) && (v_curr * v_cmd >= 0.0)) {
```

Comment:
```suggestion
  if (abs(v_cmd) >= abs(v_curr) && v_curr * v_cmd >= 0.0) {
```

Extra separators are distracting